contracts/Chaosnet.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity 0.8.9;
+
+/// @title Chaosnet
+/// @notice This is a beta staker program for stakers willing to go the extra
+/// mile with monitoring, share their logs with the dev team, and allow to more
+/// carefully monitor the bootstrapping network. As the network matures, the
+/// beta program will be ended.
+contract Chaosnet {
+  /// @notice Indicates if the chaosnet is active. The chaosnet is active
+  /// after the contract deployment and can be ended with a call to
+  /// `deactivateChaosnet()`. Once deactivated chaosnet can not be activated
+  /// again.
+  bool public isChaosnetActive;
+
+  /// @notice Indicates if the given operator is a beta operator for chaosnet.
+  mapping(address => bool) public isBetaOperator;
+
+  /// @notice Address controlling chaosnet status and beta operator addresses.
+  address public chaosnetMaestro;
+
+  event BetaOperatorsAdded(address[] operators);
+
+  event ChaosnetMaestroRoleTransferred(
+    address oldChaosnetMaestro,
+    address newChaosnetMaestro
+  );
+
+  event ChaosnetDeactivated();
+
+  constructor() {
+    _transferChaosnetMaestro(msg.sender);
+    isChaosnetActive = true;
+  }
+
+  modifier onlyChaosnetMaestro() {
+    require(msg.sender == chaosnetMaestro, "Not the chaosnet maestro");
+    _;
+  }
+
+  /// @notice Adds beta operator to chaosnet. Can be called only by the
+  /// chaosnet maestro.
+  function addBetaOperators(address[] calldata operators)
```

Comment:
I was thinking about it but it could put the contract in an inconsistent state. Say the chaosnet is active, someone has been added as a beta operator, they joined the pool, and after some time they were removed from the beta operators list. They are still in the pool though. The only solution would be to remove them from the pool, let me think more about it please.

Reply:
After some more consideration: I do not think we should add this function. The only way operator can leave the pool is by decreasing authorization to zero. There is no way for governance or anyone else to kick the operator out of the pool. I do not think we should add this feature to chaosnet either. If someone is misbehaving, the protocol should be able to deal with the misbehavior without chaosnet owner intervention. If the protocol can't deal with it, it should be fixed.